packages/core/src/kernelCoreTemplate.js

```diff
@@ -82,6 +90,43 @@
     Object.freeze(kernel)
     return kernel
 
+    function scuttleGlobalThis(extraAvoids = new Array()) {
+      let props = [...Object.getOwnPropertyNames(globalThis)]
+      if (globalThis?.Window?.prototype) {
+        props = [...props, ...Object.getOwnPropertyNames(Window.prototype)]
+      }
+      if (globalThis?.EventTarget?.prototype) {
+        props = [...props, ...Object.getOwnPropertyNames(EventTarget.prototype)]
+      }
```

Comment:
I'm not sure if the browser optimizes this. Probably not. It might be more efficient to collect the properties inside ifs into some variables and then avoiding multiple spreads on props.

Reply:
09c4b785d